llvm/lib/Transforms/Vectorize/VPlanVerifier.cpp

```diff
@@ -92,7 +92,47 @@ static bool verifyVPBasicBlock(const VPBasicBlock *VPBB,
   for (const VPRecipeBase &R : *VPBB)
     RecipeNumbering[&R] = Cnt++;
 
+  // Set of recipe types along with VPInstruction Opcodes of all EVL-related
+  // recipes that must appear at most once in Entry or Exiting blocks.
```

Comment:
Entry -> header of the vector loop region?

Drop reference to Exiting.

I think it would also be good to check if all recipes that implicitly use EVL are dominated by the EVL recipe.